Source/CarthageKit/Resolver.swift

```diff
@@ -249,6 +245,22 @@ private struct DependencyGraph: Equatable {
 
 	init() {}
 
+	/// Determines whether the set of nodes or the dependencies of them have a
+	/// dependency on the given node.
+	func edgeDependsOn(edge: Set<DependencyNode>, dependsOn: DependencyNode) -> Bool {
+		if edge.contains(dependsOn) {
+			return true
+		}
+
+		for node in edge {
+			if let nodeDependencies = self.edges[node] where edgeDependsOn(nodeDependencies, dependsOn: dependsOn) {
```

Comment:
I think it might be better to handle the nesting in `addNode`. What do you think?